1_fetch/src/get_sb_data.R

```diff
@@ -206,18 +72,95 @@ download_children <-function(sites, sb_table_reduced, dldir, workdir, outdir, ou
   
   itemfails <- as.data.frame(itemfails)
   
-  filepath1 <- file.path(outdir, out_file_name)
+  filepath1 <- paste0(outdir, "/", out_file_name, item, ".csv")
   write_csv(data_at_sites, filepath1)
-  filepath2 <- file.path(outdir, paste0("FAILS_", out_file_name))
+  filepath2 <- paste0(outdir, "/", out_file_name, item, "_FAILS.csv")
   write_csv(itemfails, filepath2)
   return_df = c(filepath1, filepath2)
   return(return_df)
 }
 
-calc_avg_dams <- function(sb_data) {
-  sb_data <- read_csv(sb_data, show_col_types = FALSE) %>%
-    suppressMessages()
-  dams <- sb_data %>%
+get_sb_data_log_g2 <- function(sb_var_ids, file_out) {
+  
+  sb_log <- tibble(sb_id = character(), 
+                   last_update = character())
+  for (i in 1:nrow(sb_var_ids)) {
+    sb_id <- as.character(sb_var_ids[i,1])
+    last_update <- item_get_fields(sb_id, "provenance")$lastUpdated
+    log_item <- tibble(sb_id, last_update)
+    sb_log <- bind_rows(sb_log, log_item)
+  }
+  
+  write_csv(sb_log, file_out)
+  return(file_out)
+  
+}
+
+prep_feature_vars <- function(sb_var_data, sites) {
+  
+  data <- sites %>%
+    select(COMID, LAT, LON, npdes, fwwd, strg, devl, cndp)
```

Comment:
These are all for the total upstream area, so I'm recommending adding the TOT prefix. Also moving the as.numeric here because I think these are all numeric columns.

```suggestion
    select(COMID, LAT, LON, npdes, fwwd, strg, devl, cndp) %>%
    rename(TOT_npdes = npdes, TOT_fwwd = fwwd, TOT_strg = strg, TOT_devl = devl, TOT_cndp = cndp) %>% 
    as.numeric()
```

Reply:
If we start renaming feature variables we might want to create a data definition file. With hundreds of different features it is hard to keep track, especially if we begin renaming. Potentially separate issue?

Reply:
Yes, good idea to have a definition file. I think a lot of these are already defined in the vars of interest xlsx file, so that's a good starting point.


These 5 variables are the only ones in the gages dataset. We could rename them in the gages xlsx file instead of renaming them here.